packages/trie/src/trie/secure.ts

```diff
@@ -29,6 +29,10 @@ export class SecureTrie extends CheckpointTrie {
    * @param value
    */
   async put(key: Buffer, val: Buffer): Promise<void> {
+    if (this._persistRoot && key.equals(ROOT_DB_KEY)) {
+      throw new Error(`Attempted to set '__root__' key but it is not allowed.`)
```

Comment:
This is technically not necessary, since in `SecureTrie` we hash the `key` before putting it in the DB. So the key can never be `__root__`

Reply:
Super nit, but in the error messages probably `ROOT_DB_KEY.toString()` should be used?